java/src/shared/model/game/Game.java

```diff
@@ -813,6 +819,15 @@ public void discardCards(int playerIndex, List<ResourceType> cards) throws Playe
         if (canDiscardCards(playerIndex) && this.turnTracker.canDiscard()) {
             playerManager.discardResourceType(playerIndex, cards);
         }
+        boolean change = true;
```

Comment:
Please use a more descriptive variable name than `change`. It is difficult to tell exactly what the purpose of this code is.